tests/testthat/test-xml_nodes_to_lints.R

```diff
@@ -86,6 +86,48 @@ test_that("it handles multi-line lints correctly", {
   expect_identical(l$ranges, list(as.integer(c(xml2::xml_attr(node, "col1"), nchar(code[1L])))))
 })
 
+test_that("it doesn't produce invalid lints", {
+  # Part of #1427
+  code <- "before   %+%   after"
+  tmpfile <- withr::local_tempfile(lines = code)
+  expr <- get_source_expressions(tmpfile)$expressions[[2L]]
+  xml <- expr$full_xml_parsed_content
+  node <- xml2::xml_find_first(xml, "//SPECIAL")
+
+  expect_warning(l_invalid_loc1 <- xml_nodes_to_lints(
```

Comment:
these tests require a lot of staring to figure out what's going on, any ideas for improvement? o/w a comment is needed